### ISSUE TITLE
Docs need updating for Lock v11

### ISSUE THREAD

**Issue by XaserAcheron** [auth0-community/ember-simple-auth-auth0#99]
Just a note-to-self to make sure the docs are accurate before release.

In particular, the "Data object on the session" section is no longer accurate since Lock v11 exclusively uses OIDC-compliant tokens.

Would probably be a good idea to add some instructions on how to make custom claims work (see #95), since that's liable to be a stumbling block for folks. Some general migration-guide stuff is a given for sure.

**Comment by XaserAcheron** [comment]
Note to self: add `autoClose` to the sample Lock options, since that's no longer set by default as of #102.

**Comment by bhushaan** [comment]
What is the estimated release date of the add on with auth0-lock v11 support ?

**Comment by dbinetti** [comment]
Hey @XaserAcheron do you need some help on the docs?  Anything we can do to help get a new release out?

**Comment by XaserAcheron** [comment]
That would be grand! I've been on a busy streak as of late and haven't been able to buckle down on this yet.

In addition to the autoClose stuff mentioned above, sections like [Data object on the session](https://github.com/auth0-community/ember-simple-auth-auth0#data-object-on-the-session) will probably need some refreshing since the tokens look different in OIDC-conformant-land.

The Passwordless stuff is pending an overhaul (Lock v11 [supports it](https://github.com/auth0/lock/pull/1070) now! woohoo!), so that's on me to give that a go-over when that gets done.

**Comment by samb0303** [comment]
👋What's the ETA on the next release?

**Comment by XaserAcheron** [comment]
Unsure at present -- I've unfortunately been busy on all fronts.

For the sake of getting some early adoption & testing in, would y'all be willing to try out a beta release if I were to cut one from _develop_ as-is? I can't guarantee a smooth transition from v3.x but it may be better than waiting on me to figure it all out.

**Comment by samb0303** [comment]
That would be great!

**Comment by dbinetti** [comment]
+1

**Comment by wuher** [comment]
Do we have the beta release? I'm getting a lot of deprecation warnings from Auth0 because of lock.js 10. April 1st is getting closer and lock 10 apparently doesn't work after that.

**Comment by XaserAcheron** [comment]
4.0.0-beta.0 has been published. Sorry for the wait.

Still some things to do -- in addition to docs, now's probably the best time to strip out the old passwordless stuff since that's been integrated into Lock v11, so expect that to be in v4.0.0 proper.

On a completely unrelated note, `npm publish` turns my terminal purple:
![cmdpurple](https://user-images.githubusercontent.com/4601589/37349418-610f40ae-26a4-11e8-8a1a-4d4a239df320.png)
Windows development is strange sometimes.

**Comment by dbinetti** [comment]
so what are the chances someone can take stab at how to use the new
library?  I started on some docs and am more confused than ever...  :-(

**Comment by XaserAcheron** [comment]
Auth0 has a [migration guide](https://auth0.com/docs/libraries/lock/v11/migration-v10-v11) for Lock v10 to v11. Are you running into any issues not covered there?

**Comment by psteininger** [comment]
@XaserAcheron I tried installing the beta v4 but I got this bunch of good news:
```
multi main
Module not found: Error: Cannot resolve module 'auth0-js/src/index.js' in /Users/piotr/lv/saas/dashboard/tmp/webpack_writer-input_base_path-jncBSCgQ.tmp/0
resolve module auth0-js/src/index.js in /Users/piotr/lv/saas/dashboard/tmp/webpack_writer-input_base_path-jncBSCgQ.tmp/0
  looking for modules in /Users/piotr/lv/saas/dashboard/node_modules
    /Users/piotr/lv/saas/dashboard/node_modules/auth0-js doesn't exist (module as directory)
[/Users/piotr/lv/saas/dashboard/node_modules/auth0-js]
 @ multi main 0 [ 'multi main\nModule not found: Error: Cannot resolve module \'auth0-js/src/index.js\' in /Users/piotr/lv/saas/dashboard/tmp/webpack_writer-input_base_path-jncBSCgQ.tmp/0\nresolve module auth0-js/src/index.js in /Users/piotr/lv/saas/dashboard/tmp/webpack_writer-input_base_path-jncBSCgQ.tmp/0\n  looking for modules in /Users/piotr/lv/saas/dashboard/node_modules\n    /Users/piotr/lv/saas/dashboard/node_modules/auth0-js doesn\'t exist (module as directory)\n[/Users/piotr/lv/saas/dashboard/node_modules/auth0-js]\n @ multi main' ]
Cannot read property 'message' of null

```

any idea how to get around that?

**Comment by XaserAcheron** [comment]
Looks like it's not pulling in the Auth0.js dependency for some reason. Would you mind double-checking that the `node_modules/auth0-js` directory exists (and npm/yarn has been run recently)?

Additionally, which version of the addon were you on previously? That may help diagnose it.

**Comment by psteininger** [comment]
That directory does not exist. I was previously on 3.1.0. I picked the beta as we really need to get on Lock 11 before April, when Lock 10 and Auth0.js 8 are hosed due to API changes

**Comment by psteininger** [comment]
@XaserAcheron shouldn't the code be looking in `node_modules/ember-simple-auth-auth0/node_modules/auth0-js` ? that's where the dependencies for this module are

**Comment by bhushaan** [comment]
@psteininger : I think, all node modules are stored in a flat structure with latest npm so I think, there wil be no node_modules folder inside ember-simple-auth-auth0. I am also using the beta version but have not got any issue so far

**Comment by XaserAcheron** [comment]
I'm not sure why 3.1.0 would work but 4.0.0-beta.0 wouldn't. 3.1.0 was the release that switched the auth0-js dependency from bower to npm, so if there were any dependency resolution shenanigans, they ought to have showed up then.

**Comment by XaserAcheron** [comment]
Noodling this a bit more, I've got an intuition on why 4.0.0-beta.0 introduced the trouble: the `auth0-lock-passwordless` dependency hasn't been ousted yet (working on that presently in a branch), and it likely wants a different version of auth0-js. This means there's two auth0-js's in the dependency tree, resulting in one of them not getting flattened.

Good news though is if that's the case, I'm shooting for getting 4.0.0-beta.1 tomorrow that kicks out the old `auth0-lock-passwordless` component in favor of Lock 11's passwordless support. That should implicitly fix the issue.

The real mystery is why it works for me. Yarn magic, I guess. :P

**Comment by psteininger** [comment]
@XaserAcheron I really appreciate you investing your time in this (an the whole project). I'm looking forward to beta.1

**Comment by XaserAcheron** [comment]
Glad to help! Apologies for cutting it so close on the timing.

Just committed some doc changes (thus technically handling this issue ;) to the `feature/nix-passwordless` branch. I'll get it merged and a new beta pushed once Travis does its thing.

**Comment by XaserAcheron** [comment]
Just published `v4.0.0-beta.1`, w/doc updates. Lemme know how it goes!

**Comment by XaserAcheron** [comment]
Hope y'all don't mind if I close this issue, since docs are done and everything seems stable-ish now. If you run into any other troubles, feel free to open a new one and I'll do my best to help get it sorted out.
